vllm/engine/arg_utils.py

```diff
@@ -3,24 +3,30 @@
 import json
 import warnings
 from dataclasses import dataclass
-from typing import List, Optional, Tuple, Union
+from typing import List, Optional, Tuple, Union, ClassVar
 
 from vllm.config import (CacheConfig, DecodingConfig, DeviceConfig,
                          EngineConfig, LoadConfig, LoRAConfig, ModelConfig,
                          ParallelConfig, SchedulerConfig, SpeculativeConfig,
                          TokenizerPoolConfig, VisionLanguageConfig)
 from vllm.model_executor.layers.quantization import QUANTIZATION_METHODS
-from vllm.utils import str_to_int_tuple
+from vllm.utils import str_to_int_tuple, deprecate_kwargs
 
 
 def nullable_str(val: str):
     if not val or val == "None":
         return None
     return val
 
-
+@deprecate_kwargs("revision",
+                is_deprecated=lambda: EngineArgs.DEPRECATE_LEGACY,
```

Comment:
Let's activate the deprecation now.

Reply:
So if I deprecate it now, I will also remove the revision I have re-added. 
Sounds reasonable?

Reply:
Let's see how the others think about this first.